tests/unit/query_helpers/test_docs_summarizer.py

```diff
@@ -125,7 +125,25 @@ def test_summarize_no_reference_content():
         llm_loader=mock_llm_loader(mock_langchain_interface("test response")())
     )
     question = "What's the ultimate question with answer 42?"
-    summary = summarizer.summarize(conversation_id, question)
+    summary = summarizer.create_response(question)
     assert question in summary.response
     assert summary.rag_chunks == []
     assert not summary.history_truncated
+
+
+@pytest.mark.asyncio
+@patch("ols.src.query_helpers.docs_summarizer.LLMChain", new=mock_llm_chain(None))
+async def test_response_generator():
+    """Test response generator method."""
+    summarizer = DocsSummarizer(
+        llm_loader=mock_llm_loader(mock_langchain_interface("test response")())
+    )
+    question = "What's the ultimate question with answer 42?"
+    summary_gen = summarizer.generate_response(question)
+    generated_content = ""
+
+    async for item in summary_gen:
+        if isinstance(item, str):
```

Comment:
just curious - does the async gen return something else than string?

Reply:
`SummarizerResponse`